wgpu-core/src/hub.rs

```diff
@@ -256,6 +278,7 @@ impl<B: hal::Backend> Access<ComputePipeline<B>> for BindGroup<B> {}
 impl<B: hal::Backend> Access<RenderPipeline<B>> for Device<B> {}
 impl<B: hal::Backend> Access<RenderPipeline<B>> for BindGroup<B> {}
 impl<B: hal::Backend> Access<RenderPipeline<B>> for ComputePipeline<B> {}
+impl<B: hal::Backend> Access<RenderPipeline<B>> for Root {}
```

Comment:
let's not modify this list please. Until we get a stronger mean of protection, we should strive to keep this minimal.
For your purposes, all of these things (`RenderPipeline`, `Buffer`, `Texture`) are accessible in the `Device` context. So you can lock the device guard for reading first, and then get any of these.